pos_self_service_weight_base/static/src/js/Screens/SelfServiceScreen/WeightWidget.js

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+odoo.define("pos_self_service_weight_base.WeightWidget", function (require) {
+    "use strict";
+
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+
+    const {onMounted, onWillUnmount, useState} = owl;
+
+    class WeightWidget extends PosComponent {
+        setup() {
+            super.setup();
+            this.state = useState({weight: 0});
+            onMounted(this.onMounted);
+            onWillUnmount(this.onWillUnmount);
+        }
+        onMounted() {
+            // Start the scale reading
+            this._readScale();
+        }
+        onWillUnmount() {
+            // Stop the scale reading
+            this.env.proxy_queue.clear();
+        }
+        _readScale() {
+            this.env.proxy_queue.schedule(this._setWeight.bind(this), {
+                duration: 500,
+                repeat: true,
+            });
+        }
```

Comment:
unlike in python, in javascript a reference to a method is not bound to a `this` variable. `bind()` returns a new function that, when called, provides the `this` variable to the method. more information [on mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind).

i’m not familiar with the `schedule()` function, but as i understand it, it will call `this._setWeight()` every 500 ms, forever.

Reply:
Thanks !